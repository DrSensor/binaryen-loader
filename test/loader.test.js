import webpack from './helpers/compiler';
import on from './helpers/on';

describe('Loader', () => {
  describe('Defaults', () => {
    const config = {
      loader: {
        test: /\.wasm$/,
      },
    };

    test('match snapshots', async () => {
      const stats = await webpack('fixture.js', config);
      on(stats).source.toMatchSnapshot();
    });

    test('indeed from rust code', async () => {
      const stats = await webpack('fixture.js', config);
      on(stats).withExtension('.rust.wasm').providedExports.toContain('rust_eh_personality');
    });

    test('not from rust code', async () => {
      const stats = await webpack('fixture.js', config);
      on(stats).withoutExtension('.rust.wasm').providedExports.not.toContain('rust_eh_personality');
    });

    test('all wasm must specify the allocated memory', async () => {
      const stats = await webpack('fixture.js', config);
      on(stats).withExtension('.wasm').providedExports.toContain('memory');
    });

    test('size must be equal or smaller', async () => {
      const stats = await webpack('fixture.js', config);
      on(stats).withExtension('.wasm').size.toBeLessThanOrEqual(_ => _.originSize);
    });
  });
});
