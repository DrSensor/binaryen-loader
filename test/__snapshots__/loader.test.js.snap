// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Defaults match snapshots 1`] = `
Object {
  "data": Array [
    0,
    97,
    115,
    109,
    1,
    0,
    0,
    0,
    1,
    7,
    1,
    96,
    2,
    127,
    127,
    1,
    127,
    3,
    7,
    6,
    0,
    0,
    0,
    0,
    0,
    0,
    5,
    3,
    1,
    0,
    1,
    7,
    67,
    7,
    3,
    97,
    100,
    100,
    0,
    0,
    8,
    115,
    117,
    98,
    116,
    114,
    97,
    99,
    116,
    0,
    1,
    8,
    109,
    117,
    108,
    116,
    105,
    112,
    108,
    121,
    0,
    2,
    6,
    100,
    105,
    118,
    105,
    100,
    101,
    0,
    3,
    5,
    112,
    111,
    119,
    101,
    114,
    0,
    4,
    9,
    114,
    101,
    109,
    97,
    105,
    110,
    100,
    101,
    114,
    0,
    5,
    6,
    109,
    101,
    109,
    111,
    114,
    121,
    2,
    0,
    10,
    49,
    6,
    7,
    0,
    32,
    0,
    32,
    1,
    106,
    11,
    7,
    0,
    32,
    0,
    32,
    1,
    107,
    11,
    7,
    0,
    32,
    0,
    32,
    1,
    108,
    11,
    7,
    0,
    32,
    0,
    32,
    1,
    109,
    11,
    7,
    0,
    32,
    0,
    32,
    1,
    115,
    11,
    7,
    0,
    32,
    0,
    32,
    1,
    111,
    11,
  ],
  "type": "Buffer",
}
`;

exports[`Loader Defaults match snapshots 2`] = `
"import unsafeArithmatic from './arithmatic.wat.wasm'; // ⬅️ size mustn't change after optimization
import matrix from './algebra_matrix2x2.rust.wasm'; // ⬅️ for testing heavy optimization
import arithmatic from './arithmatic.rust.wasm';

export default {
  unsafeArithmatic,
  matrix,
  arithmatic,
};
"
`;
